### PR TITLE
carli/2304_slow_telemetry_start

### LINKED ISSUE #2304: slow start-up of carta4 beta (2.5 minutes) when no "telemetryUuid" in .carta-beta/config/preferences.json
**Describe the bug**
When no "telemetryUuid" is in .carta-beta/config/preferences.json then the stand-alone electron version we use takes 2.5 minutes to start up. That seems to also have been the case with a remote usage without the electron app. It also happens when there is no telemetry switched on "telemetryMode": "none". Once a telemetryUuid is set, any ID, e.g. 
 "telemetryUuid": "1",
the start-up time drops to about 5 seconds. 

The process hangs on the frontend after 'compute modules loaded'. 

This may be only an issue with our set-up as reported per mail before.

**To Reproduce**
Set .carta-beta/config/preferences.json to
{
   "$schema": "https://cartavis.github.io/schemas/preferences_schema_2.json",
   "lastUsedFolder": "",
   "telemetryConsentShown": true,
   "telemetryMode": "none",
   "version": 2
}
start CARTA beta4 backend and frontend.

**Expected behavior**
Startup within 5-10 seconds

**Platform info (please complete the following information):**
 - OS RHEL8
 - electron app, but also Firefox
 - Version 4.0.0-beta.1

### PR DESCRIPTION
**Pull request by crocka**
**Description**

This PR is a simple solution that fixes #2304. ``checkAndGenerateId`` and ``flushTelemetry`` are two methods in  ``TelemetryService`` that are called during the start of the application. If the telemetry server had any problem that caused delayed response, the frontend would hang and wait for the response in the above mentioned methods, so they are altered to process asynchronously to avoid extended wait time for the users.

The two methods are studied to ensure that asynchrous call at the start of the program would not affect the telemetry performance.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] changelog updated / no changelog update needed
- [x] unit test added (for functions with no dependenies)
- [x] API documentation added (for public variables and methods in stores)

For dependencies:
- [x] e2e test passing / ~corresponding fix added~ / ~new e2e test created~
- [x] protobuf version bumped / no protobuf version bumped needed
- [x] protobuf updated to the latest dev commit / no protobuf update needed
- [x] corresponding ICD test fix added (`BackendService` changed) / no ICD test fix needed (`BackendService` unchanged)
- [x] ~user manual prepared (for large new features)~

### CODE REVIEW

**Review by YuHsuan-Hwang — APPROVED** [review]
The fix looks good.

**Comment by kswang1029** [comment]
Any tips for testing?

**Comment by kswang1029** [comment]
no regression from e2e tests. Pending manual test once I know how to.

**Comment by kswang1029** [comment]
@crocka @YuHsuan-Hwang any tip for manual testing? not sure how to test it at the moment.

**Comment by YuHsuan-Hwang** [comment]
Perhaps test with a simple server that response an error after 1 min.
<details>
  <summary>server.js</summary>

  ```js
// node server.js

const http = require('node:http');

const hostname = '127.0.0.1';
const port = 8000;
const server = http.createServer((req, res) => {
    res.setHeader('Access-Control-Allow-Origin', '*');
    if (req.url === '/api/token' && req.method === 'GET') {    
        res.writeHead(500, { 'Content-Type': 'application/json' });
        setTimeout(() => {
            res.end('Internal Server Error');
        }, 60000);
    }
});

server.listen(port, hostname, () => {
    console.log(`Server running at http://${hostname}:${port}/`);
});
  ```
</details>

File: src/services/TelemetryService.ts [hunk 1]
```diff
@@ -148,6 +148,11 @@ export class TelemetryService {
         }
 
         this.axiosInstance.defaults.headers.common = {Authorization: `Bearer ${token}`};
+
+        if (flush) {
+            yield this.flushTelemetry();
```
Comment (veggiesaurus) [inline src/services/TelemetryService.ts]: just wondering if we actually need to yield at all here, or if we can just flush the telemetry asynchronously

Comment (crocka) [reply src/services/TelemetryService.ts]: Yes, we don't need to wait for flushTelemetry(). Thank you for pointing it out!

**Review by kswang1029 — APPROVED** [review]
looks good with the simple server made by @YuHsuan-Hwang 👍 no other regression from e2e tests.

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**